test/warning.test.ts

```diff
@@ -0,0 +1,147 @@
+import { configure } from "mobx";
+import { types } from "mobx-state-tree";
+import { Field, Form, RepeatingForm, SubForm, converters } from "../src";
+
+// "strict" leads to trouble during initialization.
+configure({ enforceActions: true });
+
+test("a simple warning", async () => {
+  const M = types.model("M", {
+    foo: types.string,
+    bar: types.string
+  });
+
+  const form = new Form(M, {
+    foo: new Field(converters.string),
+    bar: new Field(converters.string)
+  });
+
+  const o = M.create({ foo: "FOO", bar: "BAR" });
+
+  const state = form.state(o, {
+    getWarning: accessor =>
+      accessor.path === "/foo" ? "Please reconsider" : undefined
+  });
+  const fooField = state.field("foo");
+  const barField = state.field("bar");
+
+  expect(fooField.raw).toEqual("FOO");
+  expect(fooField.warning).toEqual("Please reconsider");
+  expect(barField.raw).toEqual("BAR");
+  expect(barField.warning).toBeUndefined();
+  expect(state.isWarningFree).toBeFalsy();
+
+  await state.save();
+  // warnings are not cleared by a save
+  expect(fooField.warning).toEqual("Please reconsider");
+});
+
+test("a simple error", () => {
+  const M = types.model("M", {
+    foo: types.string
+  });
+
+  const form = new Form(M, {
+    foo: new Field(converters.string)
+  });
+
+  const o = M.create({ foo: "FOO" });
+
+  const state = form.state(o, {
+    getError: accessor => (accessor.path === "/foo" ? "Wrong" : undefined)
+  });
+  const fooField = state.field("foo");
+
+  expect(fooField.raw).toEqual("FOO");
+  expect(fooField.error).toEqual("Wrong");
+  expect(state.isWarningFree).toBeTruthy();
```

Comment:
you can also check that the form isn't valid, and that save returns false.

Reply:
We do want to check that 'validate' actually still returns true even if there are warnings.

Reply:
The "a simple warning" test actually verifies this